crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/validation/service/MeansAssessmentValidationService.java

```diff
@@ -1,32 +1,172 @@
+            HashMap<String, Object> uriVariables = new HashMap<>();
+            uriVariables.put(URIVAR_USERNAME, meansAssessmentRequest.getUserId());
+            uriVariables.put(URIVAR_RESERVATION_ID, meansAssessmentRequest.getReservationId());
+            uriVariables.put(URIVAR_SESSION_ID, meansAssessmentRequest.getSessionId());
+            Optional<AuthorizationResponseDTO> apiResponse = getApiResponseViaGET(validateReservationEndpoint, uriVariables, AuthorizationResponseDTO.class, meansAssessmentRequest.getLaaTransactionId());
+            if(apiResponse.isPresent()){
+                result = apiResponse.get().isResult();
+            }
+        }
+        return result;
+    }
+
+    private <T, R> Optional<R> getApiResponseViaGET(final String endpoint, final Map<String, Object> uriVariables, final Class<R> responseClass, final String laaTransactionId){
+        Mono<R> response;
+        response = webClient
+                .get()
+                .uri(endpoint, uriVariables)
+                .retrieve()
+                .bodyToMono(responseClass);
+        R responseBody = response.block();
+        if(responseClass.equals(Void.class)){
+            return Optional.empty();
+        }
+        return Optional.of(responseBody);
+    }
 
-        createAssessmentValidator.validate(assessmentValidationRequest);
+    private <T, R> Optional<R> getApiResponseViaPOST(final String endpoint, final Map<String, Object> uriVariables, final Optional<T> requestBody, final Class<R> responseClass, final String laaTransactionId){
+        Mono<R> response;
+        if(requestBody.isPresent()){
+            response = webClient
+                    .post()
+                    .uri(uriBuilder -> {
+                        return uriBuilder.build();
+                    })
+                    .bodyValue(requestBody.get())
+                    .retrieve()
+                    .bodyToMono(responseClass);
+        } else {
+            response = webClient
+                    .post()
+                    .uri(uriBuilder -> {
+                        return uriBuilder.build();
+                    })
+                    .body(BodyInserters.empty())
+                    .retrieve()
+                    .bodyToMono(responseClass);
+        }
 
+        R responseBody = response.block();
+        if(responseClass.equals(Void.class)){
+            return Optional.empty();
+        }
+        return Optional.of(responseBody);
     }
 
-    private AssessmentValidationRequest getValidationRequest(ApiCreateMeansAssessmentRequest meansAssessment) {
-        return AssessmentValidationRequest.builder()
-                .repId(meansAssessment.getRepId())
-                .userName(meansAssessment.getUserId())
-                .userAction("CREATE_ASSESSMENT")
+    @PostConstruct
+    public void initializeWebClient() {
```

Comment:
Can we not do this in the MaatApiOAuth2Client class? There is a lot of configuration being done in there and I think it makes sense to centralise it.